testdata/zk/Cheatcodes.t.sol

```diff
@@ -225,3 +225,56 @@ contract ZkCheatcodesTest is DSTest {
         // 11: EthToken
     }
 }
+
+contract UsesCheatcodes {
+    function getNonce(Vm vm, address target) public view returns (uint64) {
+       return vm.getNonce(target);
+    }
+
+    function getZkBalance(Vm vm, address target) public view returns (uint256) {
+        vm.zkVm(true);
+        getNonce(vm, target);
```

Comment:
so this gets executed in evm even though the `vm.zkVm(true)` is before this, and it would only run things in zkVm if a CALL happens from that point on right?

Reply:
Partially correct: in this instance, `getNonce` is an internal call, thus the VM is not switched before executing it, but if any call happens inside it (or we access storage translated data) then yes, it will use zkVM